src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -583,19 +586,19 @@ public bool IsDeclarationComplete(SyntaxNode decl)
         /// <summary>
         /// Marks the given symbol declaration as fully analyzed for the given analyzer.
         /// </summary>
-        public void MarkDeclarationComplete(SyntaxReference decl, DiagnosticAnalyzer analyzer)
+        public async Task MarkDeclarationCompleteAsync(SyntaxReference decl, DiagnosticAnalyzer analyzer, CancellationToken cancellationToken)
         {
-            GetAnalyzerState(analyzer).MarkDeclarationComplete(decl);
+            await GetAnalyzerState(analyzer).MarkDeclarationCompleteAsync(decl, cancellationToken).ConfigureAwait(false);
```

Comment:
await not needed. I am not marking every methods that doesnt need await. but you probably know what I am marking.